src/Sentry.Maui/Internal/MauiDeviceData.cs

```diff
@@ -64,19 +64,23 @@ public static void ApplyMauiDeviceData(this Device device, IDiagnosticLogger? lo
                 logger?.LogDebug("No permission to read network state from the device.");
             }
 
-            // https://docs.microsoft.com/dotnet/maui/platform-integration/device/display
-            var display = DeviceDisplay.MainDisplayInfo;
-            device.ScreenResolution ??= $"{(int)display.Width}x{(int)display.Height}";
-            device.ScreenDensity ??= (float)display.Density;
-            device.Orientation ??= display.Orientation switch
+#if MACCATALYST || IOS
+            if (MainThread.IsMainThread)
             {
-                DisplayOrientation.Portrait => DeviceOrientation.Portrait,
-                DisplayOrientation.Landscape => DeviceOrientation.Landscape,
-                _ => null
-            };
-            // device.ScreenDpi ??= ?
-            // ? = display.RefreshRate;
-            // ? = display.Rotation;
+                CaptureDisplayInfo();
+            }
+            else
+            {
+                // On iOS and Mac Catalyst, Accessing DeviceDisplay.Current must be done on the UI thread or else an
+                // exception will be thrown.
+                // See https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/device/display?view=net-maui-8.0&tabs=macios#platform-differences
+                var resetEvent = new ManualResetEventSlim(false);
```

Comment:
Does it need disposing?

```suggestion
                using var resetEvent = new ManualResetEventSlim(false);
```